flight/mcu/lib/teensyShared/BarometerSensor.cpp

```diff
@@ -0,0 +1,33 @@
+#include "ChRt.h"
+#include "MS5611.h"
+
+#include "BarometerSensor.h"
+#include "dataLog.h"
+
+BarometerSensor barometer;
+
+void BarometerSensor::init() {
+    MS.init();
+}
+
+void BarometerSensor::refresh() {
+    chMtxLock(&mutex);
+    MS.read(12);
+    pressure = static_cast<float>(MS.getPressure() * 0.01 + 26.03);
+    temperature = static_cast<float>(MS.getTemperature() * 0.01);
+    altitude = static_cast<float>(-log(pressure * 0.000987) * (temperature + 273.15) * 29.254);
+    dataLogger.pushBarometerFifo((BarometerData) {temperature, pressure, altitude, chVTGetSystemTime()});
+    chMtxUnlock(&mutex);
+}
+
+float BarometerSensor::getPressure() const {
+    return pressure;
+}
+
+float BarometerSensor::getTemperature() const {
+    return temperature;
+}
+
+float BarometerSensor::getAltitude() const {
+    return altitude;
+}
```

Comment:
Why make it the caller's responsibility to lock the mutex on get*, but the barometer responsibility on refresh.

Reply:
well a single pointer read is atomic by default, so making get* lock a mutex wouldn't do anything

Reply:
the problem is that reads are not guaranteed to happen unless there is some concurrency control. I think that the sensors should either ensure they are thread safe internally or ignore thread safety and let the calling code handle it in all cases. Keep it as is for this pr, we can look at changing it later.

Reply:
So we make an issue titled "Make mutexes correct"?

Reply:
yeah I guess, don't try and fix it for this pr